source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx

```diff
@@ -80,63 +84,86 @@ export const PropertySelectorSearchContainer: React.FC<IPropertySelectorSearchCo
             PIN: p?.properties?.PIN?.toString(),
           });
-        });
+        }
+        return null;
+      });
+      */
 
-        const addressResults = await Promise.all(getAddressTasks);
-        const regionDistrictResults = await Promise.all(matchTasks);
-        const pimsResults = await Promise.all(getPimsTasks);
-
-        const locations = result.features.map<SelectedFeatureDataset>((p, i) => {
-          const foundProperty = featureToLocationFeatureDataset(p);
-          foundProperty.regionFeature = regionDistrictResults[i]?.regionFeature;
-          foundProperty.districtFeature = regionDistrictResults[i]?.districtFeature;
-
-          if (exists(foundProperty?.pimsFeature)) {
-            // TODO: This needs to be changed to work with multiple properties
-            foundProperty.pimsFeature = pimsResults[i]?.features?.length
-              ? pimsResults[i]?.features[0]
-              : ({
-                  properties: {
-                    STREET_ADDRESS_1: addressResults[i]?.fullAddress,
-                  },
-                } as Feature<Geometry, PIMS_Property_Location_View>);
-          }
-          return foundProperty;
-        });
-        setSearchResults(locations);
-      } else {
-        const locations = result?.features?.map(p => featureToLocationFeatureDataset(p));
-        setSearchResults(locations ?? []);
+      // match the region and district for all found properties
+      const dataset: SelectedFeatureDataset[] = result.features.map(p =>
+        featureToLocationFeatureDataset(p),
+      );
+      const regionDistrictTasks = getRegionAndDisctricts(dataset, findRegion, findDistrict);
+      const regionDistrictResults: Dictionary<RegionDistrictResult> = {};
+
+      // Await for all results
+      for (const entry in regionDistrictTasks) {
+        const districtResult = (await regionDistrictTasks[entry].districtTask) ?? null;
+        const regionResult = (await regionDistrictTasks[entry].regionTask) ?? null;
+        regionDistrictResults[entry] = { regionResult, districtResult };
       }
+
+      /*
+       * TODO: Not used. PSP-10476
+      const pimsResults = await Promise.all(getPimsTasks);
+      const addressResults = await Promise.all(getAddressTasks);
+      */
+
+      const locations = result.features.map<SelectedFeatureDataset>(p => {
+        const foundProperty = featureToLocationFeatureDataset(p);
+        const latLngKey = propertyToLatLngKey(foundProperty);
+        if (exists(regionDistrictResults[latLngKey])) {
+          foundProperty.regionFeature = regionDistrictResults[latLngKey].regionResult;
+          foundProperty.districtFeature = regionDistrictResults[latLngKey].districtResult;
+        }
+
+        /*
+        *  TODO: this is always false. This section needs to be analized. Otherwise, both the PIMS and Address search results are not being used at all
+        * PSP-140476
+        if (exists(foundProperty?.pimsFeature)) {
```

Comment:
Note that foundProperty.pimsFeature has not been initialized at all. Line 136 creates an empty pimsFeature